lib/web/apiserver.go

```diff
@@ -351,6 +351,9 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*RewritingHandler, error) {
 	h.POST("/webapi/trustedcluster", h.WithAuth(h.upsertTrustedClusterHandle))
 	h.DELETE("/webapi/trustedcluster/:name", h.WithAuth(h.deleteTrustedCluster))
 
+	h.GET("/webapi/apps/:fqdnHint", h.WithAuth(h.getAppFQDN))
+	h.GET("/webapi/apps/:fqdnHint/:clusterName/:publicAddr", h.WithAuth(h.getAppFQDN))
```

Comment:
I would add `/resolve` action verb at the end of these to indicate these resolve app fqdn and not return actual application.

```
/webapi/apps/:fqdnHint/resolve
/webapi/apps/:fqdnHint/:clusterName/:publicAddr/resolve
```

Reply:
As it stands, this is not supported by `httprouter`: https://github.com/julienschmidt/httprouter/issues/91 (https://github.com/julienschmidt/httprouter/pull/329 is open to address this though).